packages/in_app_purchase/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -218,11 +251,30 @@ private void launchBillingFlow(
     if (accountId != null && !accountId.isEmpty()) {
       paramsBuilder.setAccountId(accountId);
     }
+    if (oldSku != null && !oldSku.isEmpty()) {
+      paramsBuilder.setOldSku(oldSku);
+    }
+    paramsBuilder.setReplaceSkusProrationMode(prorationMode);
     result.success(
         Translator.fromBillingResult(
             billingClient.launchBillingFlow(activity, paramsBuilder.build())));
   }
 
+  private int getProrationMode(final int prorationModeValue) {
```

Comment:
Should the return type be `ProrationMode` here?

Reply:
The ’ProrationMode’ is not an enum (https://developer.android.com/reference/com/android/billingclient/api/BillingFlowParams.ProrationMode)

The output of getProrationMode() is used as an argument for the method `paramsBuilder.setReplaceSkusProrationMode(prorationMode)`. Since the method `setReplaceSkusProrationMode` accepts an integer value, the return type of  `getProrationMode` is int 

At a glance, this getProrationMode() seems to be unnecessary as we can simply pass the incoming 'prorationModeValue' from flutter directly to  paramsBuilder.setReplaceSkusProrationMode(prorationModeValue). But I just wanted to explicitly mention the different proration modes. Please share the suggestions.

Reply:
I see. Yeah I think we can remove this method and put a comment above the code where we get the proration mode to link to all the proration modes:
For example:
```
// The proration mode value has to match one of the following declared in
// https://developer.android.com/reference/com/android/billingclient/api/BillingFlowParams.ProrationMode
```

Reply:
Done